File: 04 Kontrollstrukturen/Pascal/Aufgabe6.pas
Program Aufgabe6;

{$codepage utf8}

Var UserName: String;
    UserGroup: String;
    Rechtewahl: String;
    Rechte: String = '';

Begin
    Write('Bitte gib den Benutzernamen ein: ');
    ReadLn(UserName);

    Write('Bitte gib die Gruppe des Nutzers an: ');
    ReadLn(UserGroup);

    Write('Welche Rechte willst du sehen? (F - Dateirechte | P - Druckerrechte): ');
    ReadLn(Rechtewahl);

    Case UserGroup Of
        'Admins':
            If Rechtewahl = 'F' Then
                Rechte := 'Erstellen, Lesen, Löschen, Korrigieren, Rechte vergeben'
            Else If Rechtewahl = 'P' Then
                    Rechte := 'Drucken, Aufträge verwalten, Rechte vergeben';

        'PowerUsers':
            If Rechtewahl = 'F' Then
                Rechte := 'Erstellen, Lesen, Löschen, Korrigieren'
            Else If Rechtewahl = 'P' Then
                Rechte := 'Drucken, Aufträge verwalten';

        'Users':
            If Rechtewahl = 'F' Then
                Rechte := 'Erstellen, Lesen, Korrigieren'
            Else If Rechtewahl = 'P' Then
                Rechte := 'Drucken';

        'Guests':
            If Rechtewahl = 'F' Then
                Rechte := 'Lesen'
            Else If Rechtewahl = 'P' Then
                Rechte := 'Drucken';
    End;

    If (Rechtewahl <> 'F') and (Rechtewahl <> 'P') Then
        WriteLn('Es gab einen Fehler bei der Eingabe: Die Rechtekategorie ', Rechtewahl, ' existiert nicht.')
    Else If Rechte = '' Then
        WriteLn('Es gab einen Fehler bei der Eingabe: Die Gruppe ', UserGroup, ' existiert nicht.')
    Else
        WriteLn(UserName, ' (Gruppe ', UserGroup, ')', LineEnding, 'Rechte für ', Rechtewahl, ': ', Rechte);
End.

File: 04 Kontrollstrukturen/Go/Aufgabe6.go
package main

import "fmt"

var (
	userName string
	userGroup string
	rechteWahl string
	rechte string = ""
)

func main() {
	fmt.Print("Bitte gib den Benutzernamen ein: ")
	fmt.Scan(&userName)

	fmt.Print("Bitte gib die Gruppe des Nutzers an: ")
	fmt.Scan(&userGroup)
	
	fmt.Print("Welche Rechte willst du sehen? (F - Dateirechte | P - Druckerrechte): ")
	fmt.Scan(&rechteWahl)

	switch userGroup {
		case "Admins":
			if rechteWahl == "F" {
				rechte = "Erstellen, Lesen, Löschen, Korrigieren, Rechte vergeben"
			} else if rechteWahl == "P" {
				rechte = "Drucken, Aufträge verwalten, Rechte vergeben"
			}
		case "PowerUsers":
			if rechteWahl == "F" {
				rechte = "Erstellen, Lesen, Löschen, Korrigieren"
			} else if rechteWahl == "P" {
				rechte = "Drucken, Aufträge verwalten"
			}
		case "Users":
			if rechteWahl == "F" {
				rechte = "Erstellen, Lesen, Korrigieren"
			} else if rechteWahl == "P" {
				rechte = "Drucken"
			}
		case "Guests":
			if rechteWahl == "F" {
				rechte = "Lesen"
			} else if rechteWahl == "P" {
				rechte = "Drucken"
			}
	}

	if (rechteWahl != "F") && (rechteWahl != "P") {
		fmt.Println("Es gab einen Fehler bei der Eingabe: Die Rechtekategorie", rechteWahl, "existiert nicht.")
	} else if rechte == "" {
		fmt.Println("Es gab einen Fehler bei der Eingabe: Die Gruppe", userGroup, "existiert nicht.")
	} else {
		fmt.Println(userName, "(Gruppe", userGroup, ")");
		fmt.Println("Rechte für", rechteWahl, ":", rechte)
	}
}